From b69d8e8a88d73d3e7b1669cdaee4f96d5134c965 Mon Sep 17 00:00:00 2001
From: Seth Jennings <sjenning@redhat.com>
Date: Mon, 18 Aug 2014 15:42:50 -0400
Subject: [edac] add DDR4 and RDDR4

Message-id: <1408376571-8244-12-git-send-email-sjenning@redhat.com>
Patchwork-id: 87872
O-Subject: [RHEL7.1 PATCH 11/12] edac: add DDR4 and RDDR4
Bugzilla: 1127019
RH-Acked-by: Aristeu Rozanski <aris@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Aristeu Rozanski <aris@redhat.com>

Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7802951
Bugzilla: https://brewweb.devel.redhat.com/taskinfo?taskID=7802951
Upstream: 7b8278358cc2b453ca6e75eedb3741cdb7e97236

Haswell memory controller can make use of DDR4 and Registered DDR4

Cc: tony.luck@intel.com
Signed-off-by: Aristeu Rozanski <aris@redhat.com>
Signed-off-by: Mauro Carvalho Chehab <m.chehab@samsung.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/edac/edac_mc_sysfs.c b/drivers/edac/edac_mc_sysfs.c
index 88cd940..5143e5a 100644
--- a/drivers/edac/edac_mc_sysfs.c
+++ b/drivers/edac/edac_mc_sysfs.c
@@ -108,7 +108,9 @@ static const char * const mem_types[] = {
  [MEM_RDDR2] = "Registered-DDR2",
  [MEM_XDR] = "XDR",
  [MEM_DDR3] = "Unbuffered-DDR3",
- [MEM_RDDR3] = "Registered-DDR3"
+ [MEM_RDDR3] = "Registered-DDR3",
+ [MEM_DDR4] = "Unbuffered-DDR4",
+ [MEM_RDDR4] = "Registered-DDR4"
 };
 
 static const char * const dev_types[] = {
diff --git a/include/linux/edac.h b/include/linux/edac.h
index 5c6d7fb..71d6441 100644
--- a/include/linux/edac.h
+++ b/include/linux/edac.h
@@ -166,6 +166,9 @@ static inline char *mc_event_error_type(const unsigned int err_type)
  * @MEM_DDR3:  DDR3 RAM
  * @MEM_RDDR3:  Registered DDR3 RAM
  *   This is a variant of the DDR3 memories.
+ * @MEM_DDR4:  DDR4 RAM
+ * @MEM_RDDR4:  Registered DDR4 RAM
+ *   This is a variant of the DDR4 memories.
  */
 enum mem_type {
  MEM_EMPTY = 0,
@@ -185,6 +188,8 @@ enum mem_type {
  MEM_XDR,
  MEM_DDR3,
  MEM_RDDR3,
+ MEM_DDR4,
+ MEM_RDDR4,
 };
 
 #define MEM_FLAG_EMPTY  BIT(MEM_EMPTY)
-- 
1.7.1